File: pkg/server/pokete/msg/error/version_mismatch.go
package error

import "github.com/lxgr-linux/pokete/bs_rpc/msg"

const VersionMismatchType msg.Type = "pokete.error.version_mismatch"

type VersionMismatch struct {
	msg.BaseMsg
	Version string `json:"version"`
}

func (v VersionMismatch) GetType() msg.Type {
	return VersionMismatchType
}

func NewVersionMismatch(version string) VersionMismatch {
	return VersionMismatch{
		msg.BaseMsg{},
		version,
	}
}

File: src/pokete/classes/multiplayer/msg/error/version_mismatch.py
from typing import TypedDict

import pokete.bs_rpc as bs_rpc

VERSION_MISMATCH_TYPE = "pokete.error.version_mismatch"


class VersionMismatchData(TypedDict):
    version: str


class VersionMismatch(bs_rpc.Body):
    def __init__(self, data: VersionMismatchData):
        super().__init__(VERSION_MISMATCH_TYPE, data)
